Documentation/docs/source/IO.rst

```diff
@@ -0,0 +1,91 @@
+Image and Transformation IO
+***************************
+
+.. _image-io:
+
+Images
+======
+
+The default SimpleITK ``ImageFileReader`` can handle numerous file types using
+a list of registered ImageIO's. However, several ImageIO's can handle the same
+file types; if you would like to specify which IO the reader uses, use the
+``SetImageIO()`` method. A list of registered ImageIO's can be found using the 
+``GetRegisteredImageIOs()`` method, but is posted here:
+
+    - `BMPImageIO <https://itk.org/Doxygen/html/classitk_1_1BMPImageIO.html>`_ ( \*.bmp, \*.BMP )
+    - `BioRadImageIO <https://itk.org/Doxygen/html/classitk_1_1BioRadImageIO.html>`_ ( \*.PIC, \*.pic )
+    - `Bruker2dseqImageIO <https://itk.org/Doxygen/html/classitk_1_1Bruker2dseqImageIO.html>`_
+    - `GDCMImageIO <https://itk.org/Doxygen/html/classitk_1_1GDCMImageIO.html>`_
+    - `GE4ImageIO <https://itk.org/Doxygen/html/classitk_1_1GE4ImageIO.html>`_
+    - `GE5ImageIO <https://itk.org/Doxygen/html/classitk_1_1GE5ImageIO.html>`_
+    - `GiplImageIO <https://itk.org/Doxygen/html/classitk_1_1GiplImageIO.html>`_
+    - `HDF5ImageIO <https://itk.org/Doxygen/html/classitk_1_1HDF5ImageIO.html>`_
+    - `JPEGImageIO <https://itk.org/Doxygen/html/classitk_1_1JPEGImageIO.html>`_ ( \*.jpg, \*.JPG, \*.jpeg, \*.JPEG )
+    - `LSMImageIO <https://itk.org/Doxygen/html/classitk_1_1LSMImageIO.html>`_ ( \*.tif, \*.TIF, \*.tiff, \*.TIFF, \*.lsm, \*.LSM )
+    - `MINCImageIO <https://itk.org/Doxygen/html/classitk_1_1MINCImageIO.html>`_ ( \*.mnc, \*.MNC )
+    - `MRCImageIO <https://itk.org/Doxygen/html/classitk_1_1MRCImageIO.html>`_ ( \*.mrc, \*.rec )
+    - `MetaImageIO <https://itk.org/Doxygen/html/classitk_1_1MetaImageIO.html>`_ ( \*.mha, \*.mhd )
+    - `NiftiImageIO <https://itk.org/Doxygen/html/classitk_1_1NiftiImageIO.html>`_ ( \*.nia, \*.nii, \*.nii.gz, \*.hdr, \*.img, \*.img.gz )
+    - `NrrdImageIO <https://itk.org/Doxygen/html/classitk_1_1NrrdImageIO.html>`_ ( \*.nrrd, \*.nhdr )
+    - `PNGImageIO <https://itk.org/Doxygen/html/classitk_1_1PNGImageIO.html>`_ ( \*.png, \*.PNG )
+    - `StimulateImageIO <https://itk.org/Doxygen/html/classitk_1_1StimulateImageIO.html>`_
+    - `TIFFImageIO <https://itk.org/Doxygen/html/classitk_1_1TIFFImageIO.html>`_ ( \*.tif, \*.TIF, \*.tiff, \*.TIFF )
+    - `VTKImageIO <https://itk.org/Doxygen/html/classitk_1_1VTKImageIO.html>`_ ( \*.vtk )
+
+
+Example read and write:
+
+.. code-block :: python
+        
+        import SimpleITK as sitk
+
+        reader = sitk.ImageFileReader()
+        reader.SetImageIO("BMPImageIO")    #To set the ImageIO, change "BMPImageIO"
+        reader.SetFileName(inputImageFileName)
+        image = reader.Execute();
+
+        writer = sitk.ImageFileWriter()
+        writer.SetFileName(outputImageFileName)
+        writer.Execute(image)
+
+
+.. _transformation-io:
+
+Transformations
+===============
+
+In SimpleITK, transformation files are written as .tfm files. Take an example of a
```

Comment:
There are other extensions supported, by the conventional ITK text file format:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/TransformInsightLegacy/include/itkTxtTransformIO.hxx#L61

There is also the the MINCTransformIO, TransformHDF5, TransformMatlab, TransformMINC transform IO.
https://github.com/InsightSoftwareConsortium/ITK/tree/master/Modules/IO